osf/models/metaschema.py

```diff
@@ -79,3 +97,25 @@ def validate_metadata(self, metadata, reviewer=False, required_fields=False):
         except jsonschema.SchemaError as e:
             raise ValidationValueError(e.message)
         return
+
+
+class AbstractFormBlock(ObjectIDMixin, BaseModel):
```

Comment:
Is there any reason to have the `ObjectIDMixin` here? I don't really see an issue with exposing the PKs here.

Reply:
Alt: we use block_id as the ids in the API

Reply:
>Alt: we use block_id as the ids in the API

This would violate JSONAPI.

> Within a given API, each resource object’s type and id pair MUST identify a single, unique resource. 

http://jsonapi.org/format/#document-resource-object-identification

`block_id` is only guaranteed unique per schema, and some of the older ones conflict. In order to implement, we'd need to rekey any `registered_meta` with new `block_id`s.

Reply:
As far as exposing PKs, it's not a risky move, but goes against what seems to be accepted as "best practice."

That said, I don't actually have any strong objections to it.

Reply:
Hmm, I guess it would make sense to keep if we ever moved to the M2M model. Let's keep it 🤷‍♂️ .